fedimint-recurringd/src/db.rs

```diff
@@ -0,0 +1,182 @@
+use std::collections::HashMap;
+
+use fedimint_core::config::FederationId;
+use fedimint_core::core::OperationId;
+use fedimint_core::db::{AutocommitError, Database, IDatabaseTransactionOpsCoreTyped};
+use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::secp256k1::rand::thread_rng;
+use fedimint_core::secp256k1::PublicKey;
+use fedimint_core::{impl_db_lookup, impl_db_record};
+use fedimint_ln_client::recurring::{PaymentCodeId, PaymentCodeRootKey, RecurringPaymentProtocol};
+use futures::stream::StreamExt;
+use rand::Rng;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash, Ord, PartialOrd, Encodable, Decodable)]
+pub struct FederationDbPrefix([u8; 16]);
+
+impl FederationDbPrefix {
+    pub fn random() -> FederationDbPrefix {
+        FederationDbPrefix(thread_rng().gen())
+    }
+
```

Comment:
just use an increment number instead of random, you might have holes but that's fine

Reply:
Was thinking about that, would require an additional DB entry to track the counter because choosing an index and writing to the DB isn't atomic, so can't just rely on reading max existing index.

Reply:
Yes, GW uses whole federation ID as prefix and it is deduplicated in the DB. That is probably good to do here too.